scripts/update_prescriptions.py

```diff
@@ -6,93 +6,149 @@
                 resource["dispenseRequest"]["validityPeriod"]["start"] = date.today().isoformat()
                 resource["dispenseRequest"]["validityPeriod"]["end"] = (date.today() + timedelta(weeks=4)).isoformat()
 
 
-def update_prepare_examples(api_base_url, prepare_request_path, prescription_id, short_prescription_id, authored_on):
-    try:
-        with open(prepare_request_path) as f:
-            prepare_request_json = json.load(f)
-        replace_ids_and_timestamps(prepare_request_json, prescription_id, short_prescription_id, authored_on, None)
-        with open(prepare_request_path, 'w') as f:
-            json.dump(prepare_request_json, f, indent=2)
-
-        prepare_response_json = requests.post(
-            f'{api_base_url}/$prepare',
-            data=json.dumps(prepare_request_json),
-            headers={'Content-Type': 'application/json'}
-        ).json()
+def find_prepare_request_paths(examples_root_dir):
+    for filename in glob.iglob(f'{examples_root_dir}**/*Prepare-Request*200_OK*.json', recursive=True):
+        yield filename
 
-        prepare_response_path = derive_prepare_response_path(prepare_request_path)
-        with open(prepare_response_path, 'w') as f:
-            json.dump(prepare_response_json, f, indent=2)
 
-        for parameter in prepare_response_json["parameter"]:
-            if parameter["name"] == "timestamp":
-                return parameter["valueString"]
-    except BaseException as e:
-        print(f"Failed to process example at {prepare_request_path}")
-        raise e
+def load_prepare_request(prepare_request_path):
+    return load_json(prepare_request_path)
+
+
+def load_process_request(process_request_path):
+    return load_json(process_request_path)
+
+
+def load_json(path):
+    with open(path) as f:
+        return json.load(f)
+
+
+def save_prepare_request(prepare_request_path, prepare_request_json):
+    save_json(prepare_request_path, prepare_request_json)
+
+
+def save_prepare_response(prepare_response_path, prepare_response_json):
+    save_json(prepare_response_path, prepare_response_json)
+
+
+def save_process_request(process_request_path, process_request_json):
+    save_json(process_request_path, process_request_json)
+
+
+def save_convert_response(convert_response_path, convert_response_xml):
+    save_xml(convert_response_path, convert_response_xml)
+
+
+def save_json(path, body):
+    with open(path, 'w') as f:
+        json.dump(body, f, indent=2)
+
+
+def save_xml(path, body):
+    with open(path, 'w') as f:
+        f.write(body)
+
+
+def get_organisation_code(prepare_request_json):
+    # secondary-care
+    for entry in prepare_request_json["entry"]:
+        resource = entry["resource"]
+        if resource["resourceType"] == "HealthcareService":
+            for identifier in resource["identifier"]:
+                return identifier["value"]
+    # primary care
+    resources = map(getResource, prepare_request_json['entry'])
```

Comment:
Might be easier to read if you mirrored the code for HealthcareService, e.g.:
```
for entry in prepare_request_json["entry"]:
        resource = entry["resource"]
        if resource["resourceType"] == "Organization" and partOf in resource:
            for identifier in resource["identifier"]:
                return identifier["value"]
```
Up to you though

Reply:
I think we need to handle 2 use-cases: primary care with 2 orgs where one is main and also where only 1 org is given, so we're ordering the list by main org last to handle both in one, but if there's a way of writing that so it's clearer it'd be good to replace 👍

Reply:
In both cases, the more granular of the two Organizations will have a `partOf`. The `partOf` bit can can either contain the organisation details or a reference to an entire Organization resource